vochain/vochaininfo/vochaininfo.go

```diff
@@ -67,22 +63,58 @@ func (vi *VochainInfo) updateCounters() {
 
 	voteCacheSize.Set(uint64(vi.vnode.State.CacheSize()))
 	mempoolSize.Set(uint64(vi.vnode.MempoolSize()))
+	blockPeriodMinute.Set(vi.BlockTimes()[0])
+	blocksSyncLastMinute.Set(uint64(vi.BlocksLastMinute()))
 }
 
 // Height returns the current number of blocks of the blockchain.
 func (vi *VochainInfo) Height() uint64 {
 	return height.Get()
 }
 
+// averageBlockTime calculates the average block time for the last intervalBlocks blocks.
+// The timestamp information is taken from the block headers.
+func (vi *VochainInfo) averageBlockTime(intervalBlocks int64) float64 {
+	currentHeight := int64(vi.vnode.Height())
+
+	// Calculate the starting block height for the interval
+	startBlockHeight := currentHeight - intervalBlocks
+	if startBlockHeight < 0 {
+		// We cannot calculate the average block time for the given interval
+		return 0
+	}
+
+	// Fetch timestamps for the starting and current blocks
+	startTime := vi.vnode.TimestampFromBlock(startBlockHeight)
+	currentTime := vi.vnode.TimestampFromBlock(currentHeight)
+
+	if startTime == nil || currentTime == nil {
+		return 0
+	}
+
+	// Calculate the time frame in seconds
+	timeFrameSeconds := currentTime.Sub(*startTime).Seconds()
+
+	// Adjust the average block time based on the actual time frame
+	return timeFrameSeconds / float64(intervalBlocks)
+}
+
+func (vi *VochainInfo) updateBlockTimes() {
+	vi.blockTimes = [5]uint64{
+		uint64(1000 * vi.averageBlockTime(60/int64(types.DefaultBlockTime.Seconds())*1)),
```

Comment:
can `types.DefaultBlockTime.Seconds()` return 0  at some point ? i.e setting 0 to comet config for using default values ? If so, check 0 division